### PR TITLE
Onboarding progress

### PR DESCRIPTION
**Pull request by Sravanksk**

### CODE REVIEW

File: apps/workspaces/utils.py [hunk 1]
```diff
@@ -129,7 +129,9 @@ def create_or_update_general_settings(general_settings_payload: Dict, workspace_
             'sync_qbo_to_fyle_payments': general_settings_payload['sync_qbo_to_fyle_payments'],
             'map_merchant_to_vendor': map_merchant_to_vendor,
             'je_single_credit_line': general_settings_payload['je_single_credit_line'],
-            'map_fyle_cards_qbo_account': general_settings_payload['map_fyle_cards_qbo_account']
+            'map_fyle_cards_qbo_account': general_settings_payload['map_fyle_cards_qbo_account'],
+            'onboarding_state': general_settings_payload['onboarding_state']
```
Comment (Shwetabhk) [inline apps/workspaces/utils.py:133]: Add this field to workspaces table

File: apps/workspaces/apis/advanced_configurations/serializers.py [hunk 2]
```diff
@@ -115,6 +120,10 @@ def update(self, instance, validated):
 
         AdvancedConfigurationsTriggers.run_workspace_general_settings_triggers(workspace_general_settings_instance)
 
+        if instance.onboarding_state != ['COMPLETE']:
+            instance.onboarding_state = ['ADVANCED_CONFIGURATION']
+            instance.save()
+
```
Comment (Sravanksk) [inline apps/workspaces/apis/advanced_configurations/serializers.py:126]: shall I set the state to `COMPLETE` after this step? 
Or the client takes care of it?

File: apps/workspaces/apis/advanced_configurations/serializers.py [hunk 3]
```diff
@@ -68,20 +68,25 @@ class AdvancedConfigurationsSerializer(serializers.Serializer):
     general_mappings = GeneralMappingsSerializer()
     workspace_schedules = WorkspaceScheduleSerializer()
     workspace_id = serializers.SerializerMethodField()
+    onboarding_state = serializers.SerializerMethodField()
```
Comment (Shwetabhk) [inline apps/workspaces/apis/advanced_configurations/serializers.py:71]: this is not required

Comment (Shwetabhk) [reply apps/workspaces/apis/advanced_configurations/serializers.py:71]: since onboarding_state is a native field

**Review by Shwetabhk — APPROVED** [review]

**Pull request merged**